mongo/src/main/java/com/redhat/lightblue/mongo/crud/IterateAndUpdate.java

```diff
@@ -106,74 +105,71 @@ public void update(CRUDOperationContext ctx,
         int numMatched = 0;
         int numFailed = 0;
         BsonMerge merge = new BsonMerge(md);
+        BulkUpdateCtx bulkCtx=new BulkUpdateCtx(collection);
         try {
             ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.PRE_CRUD_UPDATE_RESULTSET, ctx);
             cursor = collection.find(query, null);
```

Comment:
There would have to be some reasonable (configurable?) limit to retry attempts. I would have to imagine that would be a rare scenario like some kind of DDOS attack that would cause us to hit a limit, so it probably wouldn't need to be very high (3 consecutive concurrent mods would probably be reasonable enough to give up IMHO).